service/cleanup.go

```diff
@@ -42,8 +42,9 @@ func (s *Cleanup) deleteExpiredSchedulerLocks(db *gorm.DB) {
 
 func (s *Cleanup) deleteExpiredScheduledWorkflows(db *gorm.DB) {
 	expiryTime := time.Now().Add(-s.ScheduledWorkflowTimeout)
+	fmt.Printf("Deleting scheduled workflows older than %s ...\n", expiryTime)
 
-	db.Unscoped().Model(&table.ScheduledWorkflow{}).
+	db.Model(&table.ScheduledWorkflow{}).
 		Where("updated_at < ?", expiryTime).
-		Delete(&table.ScheduledWorkflow{})
+		Unscoped().Delete(&table.ScheduledWorkflow{})
```

Comment:
because the goal is to reduce the size of the `scheduled_workflows` table, with scheduled workflows accumulating rows over time. If we do a normal `Delete` in this case because the table has no primary key and also has a `deleted_at` column, we will get the soft delete behavior, which will insert NOW in the deleted_at column, but not actually reduce the size of the table.

Reply:
https://gorm.io/docs/delete.html summarizes this behavior pretty well